final/client/src/pages/cart.tsx

```diff
@@ -4,30 +4,43 @@ import gql from 'graphql-tag';
 
 import { Header, Loading } from '../components';
 import { CartItem, BookTrips } from '../containers';
+import { RouteComponentProps } from '@reach/router';
+import { QueryResult } from '@apollo/react-common';
+import { GetCartItems } from './__generated__/GetCartItems';
 
 export const GET_CART_ITEMS = gql`
   query GetCartItems {
     cartItems @client
   }
 `;
 
-export default function Cart() {
-  const { data, loading, error } = useQuery(GET_CART_ITEMS);
+interface CartProps extends RouteComponentProps {}
+
+const Cart: React.FC<CartProps> = () => {
+  const { 
+    data, 
+    loading, 
+    error 
+  }: QueryResult<GetCartItems, any> = useQuery<GetCartItems, any>(GET_CART_ITEMS);
```

Comment:
I think you can type this hook as `const { data, loading, error } = useQuery<GetCartItems>(GET_CART_ITEMS);`. No need to declare any or specify `QueryResult`.

Reply:
Fixed.